ci/test.sh

```diff
@@ -197,6 +209,29 @@ main() {
 
         rm -rf "${td}"
     fi
+
+    # test cmake support
+    td="$(mkcargotemp -d)"
+
+    git clone \
+        --recursive \
+        --depth 1 \
+        https://github.com/cross-rs/rust-cmake-hello-world "${td}"
+
+    pushd "${td}"
+    retry cargo fetch
+    if [[ "${TARGET}" == "arm-linux-androideabi" ]]; then
+        # ARMv5te isn't supported anymore by Android, which produces missing
```

Comment:
`re2` might not be supported on some other platforms without native atomics, it's probably better to allow-list certain architectures.

Reply:
Looks like there's numerous linking issues with every musl target, likely due to a lack of locale support, but also including much more than just that. Missing symbols include:
- `__extendsftf2`
- `floor`
- `wmemcpy`
- `pthread_mutex_unlock`
- `setlocale`

Which is some soft-float intrinsics, wide character support, pthreads, and locale functions.

Reply:
Looks like the Windows GNU target issues are not our fault: re2 requires C++17 support for [shared_mutex](https://en.cppreference.com/w/cpp/thread/shared_mutex) while only providing flags for C++11 support. Let's see if we can go to an older re2 version...

https://github.com/google/re2/blob/7a65faf439295e941baa6640a717d89c1f13e9cd/CMakeLists.txt#L80-L84

Reply:
Another issue: the package we install on Ubuntu 20.04 doesn't support `std::thread`, since we have no thread support. We'll likely want to upgrade to Ubuntu 22.04 for the Windows GNU targets.

Reply:
x86_64 CentOS doesn't have `int32_t`... ok.
https://github.com/cross-rs/cross/actions/runs/3406755182/jobs/5665741079#step:12:2303

Reply:
For sufficiently-complex musl targets, it seems we need to either link to dynamic libstdc++ (see #902), or we can always link to libc (which will only provide missing symbols for the C runtime).

Adding `RUSTFLAGS="-C link-arg=-lc"` fixes the issue. @Emilgardis  thoughts on the best solution? Always link to libc, or revert #902, knowing that the latter will fail with a lot of targets? I think the latter is probably preferable, but it would require a custom musl linker that always adds `-lc`.